aiinfra-cluster/variables.tf

```diff
@@ -172,3 +172,21 @@ variable "local_dir_copy_list" {
     error_message = "All directory paths should be full path and exist on the machine. Destination path should be provided for all directory paths."
   }
 }
+
+variable "gke_node_pool_count" {
+  description = "The number of homogeneous node pools for GKE cluster."
+  type        = number
+  default     = 0
+}
+
+variable "gke_min_node_count" {
+  description = "The min node count for node pools for GKE cluster. Creation will fail if at least this number of Nodes cannot be created."
+  type        = number
+  default     = 0
+}
+
+variable "gke_max_node_count" {
+  description = "The max node count for node pools for GKE cluster. This is the actual desired number of nodes, but it may be reached eventually, not at creation time."
+  type        = number
+  default     = 0
+}
```

Comment:
Can all these numbers be zero while `orchestrator_type` is `"gke"`, or would a `validation {}` block here checking that make sense?

Reply:
Unfortunately, in the validation block we cant access other variables.

But I am planning to change these variables and make it list of node pools. That way from the provisioning tool
1. If user wants to use homogeneous NP, they can set these variables and we create homogeneous node pool for them.
2. Or they can specify node pool yaml as a string, and we use that directly to create heterogenous node pool.

I think that will address both basic and advanced customers.   
@DmitryKakurin, @stevenBorisko what do you guys think about this ?

Reply:
Added Validation and custom nodepool.